woocommerce/qit-cli
# Add support for encrypted configs

Instead of adding support for encryption, we heavily restricted the Application Password to only be able to authenticate against non-destructive QIT-related endpoints. Now, even if an attacker manages to breach into a Partner's computer and read the application password from their `$HOME` directory, it can only be used to start and view test runs.

For comparison, Composer stores GitHub Auth tokens in plain text in the `$HOME` directory, and we couldn't find no issues about this on the internet.

This PR remains open as it contains other tweaks that were implemented as part of the work done on encryption:

- When authenticating using the CD_SECRET against a non-local Manager, routes the requests through Automattic Proxy.
- Add a command to override the URL to connect to Automattic Proxy.
- Extracts the `Application` configure step into its own `bootstrap.php` file, so that we can configure an `Application` instance in unit tests.
- Code refactor, now every `Environment` has an instance of `Cache` to write and read from.
- Allow mocking remote responses on unit tests.

### How to test this:
- Spin up a local Manager instance. Make sure it has the `vendor-auth` branch.
- Make sure dev mode is enabled: `php qit-cli.php dev`
- Add an environment, connect to your local Manager
- Assert that you can start a test run, list tests and get an individual test.
- Do the same using a Partner account, using `partner:add` now to authenticate with the Application Password.
